src/langserver/main.rs

```diff
@@ -1591,6 +1654,32 @@ handle_notification! {
         return self.Initialized(_p);
     }
 
+    on Recompile(&mut self, params) {
+        let proc_name = params.arguments[0].as_str().unwrap().split("(").collect::<Vec<&str>>()[0];
+        self.show_message(MessageType::Info, format!("Compiling {}", proc_name));
+
+	    let dme = dm::detect_environment_default()
```

Comment:
weird indentation

Reply:
should be using `dm::detect_environment(&root_path, dm::DEFAULT_ENV)` like in Initialized above (handles if langserver is running out of a weird dir)